### PR TITLE
[RFC]: refactor: use "untool" dist package instead of discrete packages

### PR DESCRIPTION
**Pull request by ZauberNerd**
With this commit we replace all occurrences of @untool/* packages with
the special `untool` dist package to ensure Hops has only a single
dependency to untool.

### CODE REVIEW

**Comment by ZauberNerd** [comment]
> You've probably already discussed about it but.. I'd expect the `@untool` wrapper package to be released under `@untool` namespace (maybe `@untool/untool`). Is there a gain for having now namespaced and non-namespaced `@untool` packages?

It is a common pattern, to have some library (eg. `untool`, `lerna`, etc) be released under their canonical name, but have extra packages namespaced with `@untool/*` or `@lerna/*`.

File: packages/jest-preset/mocks/hops.js [hunk 1]
```diff
@@ -1,5 +1,5 @@
 /* eslint react/prop-types: 0 */
-const hops = require('hops/lib/runtime');
+const hops = require('untool').internal.runtime;
```
Comment (dmbch) [inline packages/jest-preset/mocks/hops.js]: Is this not somewhat breaking? After all, `hops/lib/runtime` exports quite a bit more than `untool`'s runtime...

File: packages/pwa/dom.js [hunk 2]
```diff
@@ -2,7 +2,7 @@
 
 const {
   internal: { getConfig },
-} = require('@untool/core');
+} = require('untool');
```
Comment (dmbch) [inline packages/pwa/dom.js]: Would it make sense to use `hops-config` in this case?

File: packages/pwa/lib/loader-shim.js [hunk 3]
```diff
@@ -1 +1,3 @@
-export { getConfig } from '@untool/core/lib/config';
+import { internal } from 'untool';
+
+export const getConfig = internal.getConfig;
```
Comment (dmbch) [inline packages/pwa/lib/loader-shim.js]: Would it make sense to use `hops-config` in this case?

File: packages/react/mixin.core.js [hunk 4]
```diff
@@ -13,13 +16,13 @@ class ReactCoreMixin extends Mixin {
...
     const untoolPluginIndex = jsLoaderConfig.options.plugins.findIndex(
-      isBabelPlugin('@untool/react/lib/babel')
+      isBabelPlugin(babelPlugin.path)
     );
 
     const untoolPlugin = jsLoaderConfig.options.plugins[untoolPluginIndex];
 
     jsLoaderConfig.options.plugins[untoolPluginIndex] = [
-      require.resolve('@untool/react/lib/babel'),
+      require.resolve(babelPlugin.path),
```
Comment (dmbch) [inline packages/react/mixin.core.js]: The `require.resolve` call appears to be redundant.

File: packages/jest-preset/mocks/hops.js (hunk 1 above)
Comment (ZauberNerd) [reply packages/jest-preset/mocks/hops.js]: Yes, you're right. I've reverted this change.

**Comment by ZauberNerd** [comment]
@dmbch @toomuchdesign please take another look

**Comment by ZauberNerd** [comment]
We decided to drop this PR, because otherwise we would have to forward all peer dependencies of `untool` (`react`, `react-dom`, `react-helmet` and `react-router-dom`) to all "top level" packages of Hops (basically all Hops packages would need to forward all other peer dependencies) and this would grow worse for our internal packages as well.
Also, we do have the doctor in place to warn about duplicate packages, so we feel a bit more confident, that we will catch misaligned dependencies anyways.

**Pull request closed without merging**